src/test/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScoreTest.java

```diff
@@ -239,12 +243,138 @@ public void testFourteenMonthOldSingleVersion() {
     ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
     Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(
         ArtifactVersions.of(version100));
-    ScoreValue value = score.calculate(versions);
-    Assert.assertEquals(0.0, value.get(), DELTA);
+    ScoreValue value = score.calculate(versions, ARTIFACT_VERSION.value(version100));
+    assertEquals(0.0, value.get(), DELTA);
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void testWithNoInfo() {
     new ArtifactReleaseHistoryScore().calculate();
   }
-}
+
+  @Test
+  public void testVersionsSelectionByComparingMajor() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(7 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(6 * 31));
+    ArtifactVersion version120 = new ArtifactVersion("1.2.0", now.minusDays(5 * 31));
+    ArtifactVersion version130 = new ArtifactVersion("1.3.0-MIGHTY", now.minusDays(4 * 31));
+    ArtifactVersion version140 = new ArtifactVersion("1.4.0", now.minusDays(3 * 31));
+    ArtifactVersion version150 = new ArtifactVersion("1.5.0", now.minusDays(64));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(31));
+    ArtifactVersions artifactVersions = ArtifactVersions.of(version100, version110, version120,
+        version130, version140, version150, version200);
+
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(artifactVersions);
+    Value<ArtifactVersion> version = ARTIFACT_VERSION.value(version120);
+    ScoreValue value = score.calculate(versions, version);
+    assertEquals(8.8, value.get(), DELTA);
```

Comment:
Could you please add `assertEquals()` for "2.0.0", "1.5.0" and "1.0.0"?

Reply:
I assumed you were talking about this `assertTrue(value.usedValues().contains(versions));`
Basically, check if the entered artifact versions are part of used values.

Now it is now clear to me. @artem-smotrakov Could you please elaborate a bit more?

Reply:
Please calculate scores for "2.0.0", "1.5.0" and "1.0.0" and check if they are expected.